iframe.html

```diff
@@ -44,7 +44,13 @@
           }
           if (typeof currentLevel === 'string')
             throw("Exception: Invalid query parameter");
-          currentLevel[keySplits[0]] = decodeURIComponent(match[2]);
+
+          if (decodeURIComponent(match[2]).toLowerCase() === "true")
+            currentLevel[keySplits[0]] = true;
+          else if (decodeURIComponent(match[2]).toLowerCase() === "false")
+            currentLevel[keySplits[0]] = false;
+          else
+            currentLevel[keySplits[0]] = decodeURIComponent(match[2]);
```

Comment:
At the point we get to the if statements, currentLevel should be the object we're writing a key/value pair to, so I switched that to make sure that currentLevel is an object to be more concise. Then keySplits[0] is the last token from the query param and match[2] is the value of from the query param.